### ISSUE TITLE
Inconsistencies in Array64 creation methods

### ISSUE THREAD

**Issue by pinpox** [Kunde21/numgo#2]
Hi, 
I have been working with your package a while and noticed a few inconsistencies in the signatures of the functions used to create different types of arrays.

Consider these three functions:
```go
func NewArray64(data []float64, shape ...int) (a *Array64)
func Full(val float64, shape ...int) (a *Array64)
func RandArray64(base, scale float64, shape []int) (a *Array64)
```

- Arguments
All Functions accept variadic inputs. Why is RandArray asking for a slice instead?

- Naming:
Why is the function for filled arrays not called FullArray64, like the others?

If you are iinterested in changing this, I would submit a pull request.
My proposal would be:

```go
func NewArray64(data []float64, shape ...int) (a *Array64)
func NewFullArray64(val float64, shape ...int) (a *Array64)
func NewRandArray64(base, scale float64, shape ...int) (a *Array64)
```

Let me know what you think!

**Comment by Kunde21** [comment]
Both Full and Rand started as convenience functions while I was testing, so cleaning them up got a bit ignored along the way.  

On your suggestions:
- I agree `RandArray64` should be variadic and `Full` should be `FullArray64`.  

- I'm not sold on prepending "New" onto the `FullArray64` and `RandArray64` constructors.  The function signatures make it fairly obvious that they're all constructors, so I don't feel that adding `New` will help.  It also seems to bury the operations (full/rand) in the middle of the function name.

**Comment by pinpox** [comment]
That seems reasonable. Let's just append the "Array64"  then.

**Comment by Kunde21** [comment]
Changes merged in #4
